assert/assertions_test.go

```diff
@@ -2147,24 +2147,38 @@ func ExampleComparisonAssertionFunc() {
 	}
 }
 
+type Foo struct {
+	Foo string
+	Bar string
+}
+
+func (foo *Foo) Name() string {
+	return foo.Foo + " " + foo.Bar
+}
+
 func TestComparisonAssertionFunc(t *testing.T) {
 	type iface interface {
 		Name() string
 	}
 
+	t2 := &Foo{
+		Foo: "foo",
+		Bar: "bar",
+	}
+
 	tests := []struct {
 		name      string
 		expect    interface{}
 		got       interface{}
 		assertion ComparisonAssertionFunc
 	}{
-		{"implements", (*iface)(nil), t, Implements},
-		{"isType", (*testing.T)(nil), t, IsType},
-		{"equal", t, t, Equal},
-		{"equalValues", t, t, EqualValues},
-		{"notEqualValues", t, nil, NotEqualValues},
-		{"exactly", t, t, Exactly},
-		{"notEqual", t, nil, NotEqual},
+		{"implements", (*iface)(nil), t2, Implements},
+		{"isType", (*Foo)(nil), t2, IsType},
+		{"equal", t2, t2, Equal},
+		{"equalValues", t2, t2, EqualValues},
+		{"notEqualValues", t2, nil, NotEqualValues},
+		{"exactly", t2, t2, Exactly},
+		{"notEqual", t2, nil, NotEqual},
```

Comment:
go-cmp cannot compare `*testing.T` due to unexported fields, so tests were changed to use `t2`.